### PR TITLE
add document for hpa with custom metric

### PR DESCRIPTION
**Pull request by ngtuna**
I am adding a document to explain the setup for autoscaling with custom metric which have been tested on k8s 1.8 (with kubeadm). The setup is motivated by the [Lucas's workshop](https://github.com/luxas/kubeadm-workshop#deploying-a-custom-metrics-api-server-and-a-sample-app).

cc/ @luxas @DirectXMan12

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6109)

### CODE REVIEW

**Comment by DirectXMan12** [comment]
This is specific to Prometheus.  We decided a while ago in SIG Instrumentation that we weren't going to endorse any metrics pipeline, so I don't think we should start here (providing official InfluxDB images, IMO, is one of the issues that got us into the maintenance mess that we're in with Heapster, IMO).

Please *do not* merge this without further discussion.

**Comment by zacharysarah** [comment]
@DirectXMan12 @luxas 👋 Any consensus on whether to approve or deny this PR?

**Comment by zacharysarah** [comment]
@DirectXMan12 👋 Heya, I'm a SIG-docs maintainer. 

TLDR: No, but.

There's existing precedent for third-party-specific documentation. (For example: all of the pages for installing [network provider policies](https://kubernetes.io/docs/tasks/administer-cluster/calico-network-policy/).) Precedent notwithstanding, I wish we didn't publish third-party specific docs. They've been problematic for exactly the reason @luxas points out: ownership of ongoing maintenance has been sporadic.

That said, in this case I'm happy to link to an adapter's own documentation. @ngtuna, this is a great start! 🥇 I think this PR can be merged with some work. Specifically: 
1. Extract Prometheus-specific content from steps 3 and 4.
2. Rework steps 3 and 4 to be more broadly applicable to any adapter.

TLDR: No, but.

**Comment by DirectXMan12** [comment]
@zacharysarah thanks!  Seems like we're on the same page.

**Comment by chenopis** [comment]
@ngtuna FYI, Travis is returning this error:

> Error: docs/tasks/run-application/horizontal-pod-autoscale-custom-metric-walkthrough.md doesn't have an entry in the table of contents under _data/*.yml
Found 1 files without entries. For how to fix it, see http://kubernetes.io/docs/home/contribute/write-new-topic/#creating-an-entry-in-the-table-of-contents

So you will need to add `docs/tasks/run-application/horizontal-pod-autoscale-custom-metric-walkthrough.md ` to the `/_data/tasks.yml` file so that it's somewhere in the table of contents.

**Comment by ngtuna** [comment]
Thanks for feedbacks everyone. I updated PR for the `_data/tasks.yml`, thanks @chenopis 👍 

> That said, in this case I'm happy to link to an adapter's own documentation. @ngtuna, this is a great start! 🥇 I think this PR can be merged with some work. Specifically:
>
> 1. Extract Prometheus-specific content from steps 3 and 4 and host it elsewhere.
> 2. Rework steps 3 and 4 to be more broadly applicable to any adapter.
> 3. Include a link to the hosted content.

afaik, you guys don’t want to mention Prometheus and Prometheus-Operator in the doc, and expect to have the autoscaling with custom metric works with any adapter (heapster for example). Do I understand it right ?

**Comment by zacharysarah** [comment]
@ngtuna 👋 
> afaik, you guys don’t want to mention Prometheus and Prometheus-Operator in the doc

I'm OK with mentioning it, as long as it's a link to content that Prometheus (for example) hosts elsewhere.

> and expect to have the autoscaling with custom metric works with any adapter (heapster for example).

For documentation purposes, we want to keep the docs as specific to Kubernetes function as possible. I can't speak to "works with any adapter", but for adapters that _do_ work, we want to link to product-specific docs hosted elsewhere, rather than take on the maintenance burden of including them in the Kubernetes doc set.

Does that help?

**Comment by ngtuna** [comment]
@zacharysarah Thanks for the clarify 👍 Will work on the update shortly

**Comment by luxas** [comment]
@ngtuna let us know when this is ready for review

**Comment by ngtuna** [comment]
@luxas Sorry I haven't had time to work on this update. I guess the document is out-dated a bit. Next week I have some days off work so I will have chance to catch up and update this properly. Some folks got back to me that GKE 1.9 might want to support custom metric HPA.

**Comment by tengqm** [comment]
:+1:

**Comment by zacharysarah** [comment]
@ngtuna Please feel free to re-open this PR when you're ready to continue with it. 👍
